src/js/components/products/detail.tsx

```diff
@@ -32,7 +45,14 @@ const ProductDetail = ({ product, productSlug, doFetchObject }: Props) => {
         filters: { slug: productSlug },
       });
     }
-  }, [product, productSlug, doFetchObject]);
+
+    if (product) {
```

Comment:
I don't think we want to do this every time the product changes... so we'll probably want to check and see if we've already fetched the projects for this product? E.g. once you have the `projects` state in this component, something like:

```js
if (product && !projects[product.id]) { ... }
```